includes/sanitizers/class-amp-style-sanitizer.php

```diff
@@ -1754,6 +1789,40 @@ private function finalize_styles() {
 		}
 	}
 
+	/**
+	 * Convert CSS selectors.
+	 *
+	 * @param DeclarationBlock $ruleset Ruleset.
+	 */
+	private function ampify_ruleset_selectors( $ruleset ) {
+		$selectors    = array();
+		$replacements = 0;
+		foreach ( $ruleset->getSelectors() as $old_selector ) {
+			$edited_selectors = array( $old_selector->getSelector() );
+			foreach ( $this->selector_mappings as $html_selector => $amp_selectors ) { // Note: The $selector_mappings array contains ~6 items.
```

Comment:
These notes are here because 4 level of loop nesting is normally something that is a big red flag for performance.